pkg/router/template/configmanager/haproxy/manager.go

```diff
@@ -1105,6 +1111,16 @@ func modAnnotationsList(termination routev1.TLSTerminationType) []string {
 		"router.openshift.io/haproxy.health.check.interval",
 	}
 
+	// allow adding list of extra annotations for HAProxy blueprint
+	listExtraAnnotations, ok := os.LookupEnv("EXTRA_ANNOTATIONS_FOR_BLUEPRINT")
+	if ok {
+		option := strings.Split(listExtraAnnotations, ",")
+		for _, elem := range option {
+			glog.V(4).Infof("EXTRA_ANNOTATIONS_FOR_BLUEPRINT has anotation %s", elem)
+			annotations = append(annotations, elem)
+		}
```

Comment:
We should check for empty strings/trim spaces since this is user input. 

Something like: 
```
for _, entry := range strings.Split(listExtraAnnotations, ",") {
   if v := strings.Trim(entry); len(v) > 0 {
        annotations = append(annotations, v)
   }
}
```

Or alternatively we can do build this up just once and save it the `haproxyConfigManager` instance - maybe in `NewHAProxyConfigManager`? 
You'd have to pass the value from `cm.customAnnotations` (example name) down to this function from the calling layers.